File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/setting/ui/SecessionResultFragment.java
package com.pplus.prnumberbiz.apps.setting.ui;


import android.graphics.Typeface;
import android.os.Bundle;
import androidx.core.content.ContextCompat;
import android.text.SpannableString;
import android.text.Spanned;
import android.text.TextPaint;
import android.text.method.LinkMovementMethod;
import android.text.style.ClickableSpan;
import android.view.View;
import android.widget.TextView;

import com.pplus.prnumberbiz.R;
import com.pplus.prnumberbiz.apps.common.ui.base.BaseFragment;
import com.pplus.prnumberbiz.core.util.PplusCommonUtil;

import java.text.SimpleDateFormat;
import java.util.Date;

/**
 * 회원 탈퇴 마지막 호출
 */
public class SecessionResultFragment extends BaseFragment{

    public static SecessionResultFragment newInstance(){

        SecessionResultFragment fragment = new SecessionResultFragment();
        Bundle args = new Bundle();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState){

        super.onCreate(savedInstanceState);
//        if(getArguments() != null) {
//            mTokenReturnTime = getArguments().getString(Const.TOKEN_RETURN_TIME);
//        }
    }

    @Override
    public String getPID(){

        return "";
    }

    @Override
    public int getLayoutResourceId(){

        return R.layout.fragment_secession_result;
    }

    @Override
    public void initializeView(View container){

        TextView textDate = (TextView) container.findViewById(R.id.text_secessionDate);
        Date todayDate = new Date(System.currentTimeMillis());
        SimpleDateFormat format = new SimpleDateFormat("yyyy.MM.dd");
        String date = format.format(todayDate);
        textDate.setText(getActivity().getString(R.string.word_secessionDate) + date);
        TextView textDescription = (TextView) container.findViewById(R.id.text_secession_description2);
        SpannableString ss = new SpannableString(getString(R.string.msg_secession_description2));

        ClickableSpan clickableSpan = new ClickableSpan(){

            @Override
            public void updateDrawState(TextPaint ds){

                ds.setUnderlineText(true);
                ds.setColor(ContextCompat.getColor(getActivity(), R.color.color_ad00cc));
                ds.setTypeface(Typeface.create(Typeface.DEFAULT, Typeface.BOLD));
            }

            @Override
            public void onClick(View view){

                PplusCommonUtil.Companion.logOutAndRestart();
            }
        };
        ss.setSpan(clickableSpan, 17, 22, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
        textDescription.setText(ss);
        textDescription.setMovementMethod(LinkMovementMethod.getInstance());
        container.findViewById(R.id.text_authConfirm).setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View view){

                PplusCommonUtil.Companion.logOutAndRestart();
            }
        });
    }

    @Override
    public void init(){

    }

    public SecessionResultFragment(){
        // Required empty public constructor
    }


}

File: PRNumber/app/src/main/java/com/pplus/prnumberuser/apps/setting/ui/SecessionResultFragment.kt
package com.pplus.prnumberuser.apps.setting.ui

import android.os.Bundle
import android.text.SpannableString
import android.text.Spanned
import android.text.TextPaint
import android.text.method.LinkMovementMethod
import android.text.style.ClickableSpan
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import com.pplus.prnumberuser.R
import com.pplus.prnumberuser.apps.common.ui.base.BaseActivity
import com.pplus.prnumberuser.apps.common.ui.base.BaseFragment
import com.pplus.prnumberuser.core.util.PplusCommonUtil.Companion.logOutAndRestart
import com.pplus.prnumberuser.databinding.FragmentSecessionResultBinding
import java.text.SimpleDateFormat
import java.util.*

/**
 * 회원 탈퇴 마지막 호출
 */
class SecessionResultFragment : BaseFragment<BaseActivity>() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun getPID(): String? {
        return ""
    }

    private var _binding: FragmentSecessionResultBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {

        _binding = FragmentSecessionResultBinding.inflate(inflater, container, false)
        val view = binding.root
        return view
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    override fun init() {
        val todayDate = Date(System.currentTimeMillis())
        val format = SimpleDateFormat("yyyy.MM.dd")
        val date = format.format(todayDate)
        binding.textSecessionDate.text = getString(R.string.word_secessionDate) + date
        val ss = SpannableString(getString(R.string.msg_secession_description2))
        val clickableSpan: ClickableSpan = object : ClickableSpan() {
            override fun updateDrawState(ds: TextPaint) {
                ds.isUnderlineText = true
                ds.color = ContextCompat.getColor(activity!!, R.color.color_579ffb)
            }

            override fun onClick(view: View) {
                logOutAndRestart()
            }
        }
        ss.setSpan(clickableSpan, 14, 21, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE)
        binding.textSecessionDescription2.text = ss
        binding.textSecessionDescription2.movementMethod = LinkMovementMethod.getInstance()
        binding.textAuthConfirm.setOnClickListener { logOutAndRestart() }

    }

    companion object {
        @JvmStatic
        fun newInstance(): SecessionResultFragment {
            val fragment = SecessionResultFragment()
            val args = Bundle()
            fragment.arguments = args
            return fragment
        }
    }
}